detanalysis/semiautocuts.py

```diff
@@ -1825,31 +1895,125 @@ def combine_cuts(self, sat_pass_threshold=None, cut_name=None):
+            current_cut_arr = self.df[cuts_to_combine[i]].values
             cuts_all_arr = np.logical_and(cuts_all_arr, current_cut_arr)
-        
+            if lgc_diagnostics:
+                print("Done with cut " + str(i))
+            
             i += 1
             
+            
         if sat_pass_threshold is not None:
             ofamps_arr_all = self.df[self.ofamp_rq].values
             ofamp_thresh_mask = (ofamps_arr_all > sat_pass_threshold)
             
             cuts_all_arr = np.logical_or(cuts_all_arr, ofamp_thresh_mask)
+            if lgc_diagnostics:
+                print("Passed saturated events above " + str(sat_pass_threshold))
         
         if cut_name is None:
             self.cut_name = 'cut_all_' + self.channel_name
         self.df[self.cut_name] = cuts_all_arr
             
         self.mask = cuts_all_arr
+        
+    def create_load_combine_cuts(self, master_dict, lgc_diagnostics=False):
+        """
+        Using the previous three functions, creates new cut RQs,
+        loads cut levels, and combines relevent cuts into a master
+        cut.
+        
+        Parameters
+        ----------
+        
+        master_dict : dict of dicts
+            Dictionary of dictionaries, with keys:
+            -'creation': returns the list of dicts used to create RQs
+            -'cuts': returns the list of dicts used to create the cuts
+            -'combo': a dict with parameters 'cut_name' and 
+             'sat_pass_threshold' for the cut combination.
+             'cuts_to_combine' is an optional list of cuts to combine, if
+             they're not being loaded throigh the 'cuts' dict within the
+             master_dict.
+            
+        lgc_diagnostics : bool, optional
+            If True, prints out diagnostic statements
+             
+        """
+        
+        creation = master_dict['creation']
+        self.create_cut_rqs(creation, lgc_diagnostics=lgc_diagnostics)
+        if lgc_diagnostics:
+            print("Done with RQ creation")
+            print(" ")
+        
+        cuts = master_dict['cuts']
+        self.load_cut_dicts(cuts, lgc_diagnostics=lgc_diagnostics)
+        if lgc_diagnostics:
+            print("Done with cut creation")
+            print(" ")
+        
+        combo = master_dict['combo']
+        if 'cuts_to_combine' in combo.keys():
+            cuts_to_combine = combo['cuts_to_combine']
+        else:
+            cuts_to_combine = None
+            
+        self.combine_cuts(sat_pass_threshold=combo['sat_pass_threshold'],
+                          cut_name=combo['cut_name'],
+                          cuts_to_combine=cuts_to_combine,
+                          lgc_diagnostics=lgc_diagnostics)
+        if lgc_diagnostics:
+            print("Done with cut combination")
+            print(" ")
+                          
+    def batch_create_load_combine_cuts(self, master_dict_list, lgc_diagnostics=False):
```

Comment:
Is there a reason to have this and the previous function separate? Seems like you would always just use the batch one explicitly, so why not just add the for loop to the previous function to consolidate?

Reply:
I was taught it was better convention to separate things into two simple functions rather than one complex one... so mostly just a style thing.